### ISSUE TITLE
Mixing Express and React

### ISSUE THREAD

**Issue by dustinblackman** [mozilla-neutrino/neutrino-dev#300]
Neutrino looks fantastic, especially for devs who find themselves not tabling in web development often enough to keep up with all the major changes. I'm finding trouble in mixing presets however. The end goal I'm trying to accomplish is building an React app with Express to support server side rendering and other need endpoints. The following didn't work as expected, and ended up with a crash due to missing files.

__.neutrinorc.js__
```javascript
const react_config = {
  hot: true
};

const node_config = {
  options: {
    source: 'server',
    output: 'dist-server'
  }
};

module.exports = {
  use: [
    ['neutrino-preset-node', node_config]
    ['neutrino-preset-react', react_config],
    'neutrino-middleware-style-loader'
  ]
};
```

What's the correct approach? I'd be happy to write some docs after getting a working solution going.

**Comment by constgen** [comment]
These presets are not designed to be used in one project. Currently we don't have any SSR preset. It should not look so simple as in your example. It is a quite complex task. I am afraid I can't suggest you anything right now. Somebody just need to develop a server side rendering middleware for React.

**Comment by eliperelman** [comment]
The reason why this is easier in Webpack but nothing exists in Neutrino yet is because we haven't really found a good solution for spinning up 2 types of targets. Webpack achieves this by letting you pass an array of configs to the Webpack API:

```js
webpack([
  reactConfig,
  expressConfig
])
```

The reason why this has proven harder in Neutrino is the composition of configs doesn't really have a concept of targeting a particular config. I may have an idea for this; I'll post it in another issue.

**Comment by eliperelman** [comment]
See #308.

**Comment by eliperelman** [comment]
Being addressed by #425.

**Comment by eliperelman** [comment]
Addressed in #425.
